app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java

```diff
@@ -70,13 +106,70 @@ protected void onCreate(Bundle savedInstanceState) {
             Log.e(TAG, e.toString());
         }
         eventRegisterEventAddressTV.setText(address);
+        eventRegisterStatusTV.setText("You are not registered for this event");
+
+        reference = database.getReference("Volunteers").child(mUser.getUid());
+        reference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NotNull DataSnapshot dataSnapshot) {
+                if (dataSnapshot.child("Events").child(eventID).getValue() != null) {
+                    eventRegisterButton.setVisibility(View.GONE);
+                    if (dataSnapshot.child("Events").child(eventID).getValue().toString().equals("pending") || dataSnapshot.child("Events").child(eventID).getValue().toString().equals("registered")) {
+                        eventRegisterStatusTV.setText("You have registered for this event");
+                        eventCancelButton.setVisibility(View.VISIBLE);
+                    }
+                    if (dataSnapshot.child("Events").child(eventID).getValue().toString().equals("cancelled")) {
```

Comment:
rather than doing `dataSnapshot.child("Events").child(eventID)` every time here, maybe just go to the child from reference.
ie:
`reference.child("Events").child(eventID).addListener.......`
then you can reference to event by just doing `datasnapshot.getValue()`

Reply:
Fixed with latest commit